src/acceptance-test/java/de/zalando/aruha/nakadi/repository/db/EventTypeDbRepositoryTest.java

```diff
@@ -0,0 +1,162 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.NakadiException;
+import de.zalando.aruha.nakadi.config.NakadiConfig;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.domain.EventTypeSchema;
+import de.zalando.aruha.nakadi.repository.DuplicatedEventTypeNameException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import de.zalando.aruha.nakadi.repository.NoSuchEventTypeException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNull.notNullValue;
+
+public class EventTypeDbRepositoryTest {
+
+    private DataSource datasource;
+    private JdbcTemplate template;
+    private EventTypeRepository repository;
+    private Connection connection;
+
+    private static final String postgresqlUrl = "jdbc:postgresql://localhost:5432/local_nakadi_db";
+    private static final String username = "nakadi_app";
+    private static final String password = "nakadi";
+
+    @Before
+    public void setUp() {
+        try {
+            ObjectMapper mapper = (new NakadiConfig()).jacksonObjectMapper();
+
+            datasource = new DriverManagerDataSource(postgresqlUrl, username, password);
+            template = new JdbcTemplate(datasource);
+            repository = new EventTypeDbRepository(template, mapper);
+            connection = datasource.getConnection();
+
+            template.execute("DELETE FROM zn_data.event_type");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenCreateNewEventTypePersistItInTheDatabase() throws Exception {
+        EventType eventType = buildEventType();
+
+        repository.saveEventType(eventType);
+
+        final int rows = template.queryForObject("SELECT count(*) FROM zn_data.event_type", Integer.class);
+        assertThat("Number of rows should encrease", rows, equalTo(1));
+
+        SqlRowSet rs = template.queryForRowSet("SELECT et_name, et_event_type_object FROM zn_data.event_type");
+        rs.next();
+
+        assertThat("Name is persisted", rs.getString(1), equalTo("event-name"));
+
+        ObjectMapper mapper = (new NakadiConfig()).jacksonObjectMapper();
+        EventType persisted = mapper.readValue(rs.getString(2), EventType.class);
+
+        assertThat(persisted.getCategory(), equalTo(eventType.getCategory()));
```

Comment:
Actually I would like to "deep compare" object properties. I know that apache commons has something for that. Do you have any suggestions? The problem with this code is that once someone adds a new property, this test will be "less complete".